sklearn/tree/export.py

```diff
@@ -55,32 +142,137 @@ def export_graphviz(decision_tree, out_file="tree.dot", feature_names=None,
     >>> tree.export_graphviz(clf,
     ...     out_file='tree.dot')                # doctest: +SKIP
     """
+
+    def get_colour(value):
```

Comment:
That function reaches into the closure's scope, I could have the dicts of pre-calculated colours/bounds bounce around between functions, but a bit messier?